src/st/lib/e_shop.c

```diff
@@ -0,0 +1,691 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+#include "lib.h"
+
+/// the first 5 are the same as `EquipKind`.
+/// `INVENTORY_RELIC` and `INVENTORY_DOCUMENT` are
+/// special.
+typedef enum {
+    INVENTORY_HAND,
+    INVENTORY_HEAD,
+    INVENTORY_BODY,
+    INVENTORY_CAPE,
+    INVENTORY_ACCESSORY,
+    INVENTORY_RELIC,
+    INVENTORY_DOCUMENT,
+} InventoryCategory;
+
```

Comment:
The bulk of new work is at the top of `e_shop.c`. The rest is either updating fields and constants in `func_us_801B2BE4` or moving functions to new segments.